pkg/vault/create/create_integration_test.go

```diff
@@ -0,0 +1,251 @@
+			Expect(versionsDir).Should(BeADirectory())
+
+			resolver := versionstream.VersionResolver{
+				VersionsDir: versionsDir,
+			}
+			imageMap, err := vaultCreatorUnderTest.dockerImages(resolver)
+			Expect(err).Should(BeNil())
+
+			Expect(imageMap).Should(HaveKeyWithValue("banzaicloud/bank-vaults", "banzaicloud/bank-vaults:0.5.3"))
+			Expect(imageMap).Should(HaveKeyWithValue("vault", "vault:1.2.3"))
+		})
+
+		It("should return images unresolved if versions are missing", func() {
+			emptyDir, err := ioutil.TempDir("", "jx-create-integration-test")
+			Expect(err).Should(BeNil())
+			defer func() {
+				_ = os.RemoveAll(emptyDir)
+			}()
+
+			resolver := versionstream.VersionResolver{
+				VersionsDir: emptyDir,
+			}
+			imageMap, err := vaultCreatorUnderTest.dockerImages(resolver)
+			Expect(imageMap).Should(HaveKeyWithValue("banzaicloud/bank-vaults", "banzaicloud/bank-vaults"))
+			Expect(imageMap).Should(HaveKeyWithValue("vault", "vault"))
+		})
+	})
+
+	Describe("#CreateOrUpdateVault", func() {
+		var (
+			projectID   string
+			clusterName string
+			zone        string
+			resolver    versionstream.VersionResolver
+		)
+
+		BeforeEach(func() {
+			data, err := kube.ReadInstallValues(kubeClient, devNamespace)
+			Expect(err).Should(BeNil())
+
+			projectID = data[kube.ProjectID]
+			Expect(projectID).ShouldNot(BeEmpty())
+
+			clusterName = data[kube.ClusterName]
+			Expect(clusterName).ShouldNot(BeEmpty())
+
+			zone = data[kube.Zone]
+			Expect(zone).ShouldNot(BeEmpty())
+
+			versionsDir := path.Join("test_data", "jenkins-x-versions")
+			Expect(versionsDir).Should(BeADirectory())
+			resolver = versionstream.VersionResolver{
+				VersionsDir: versionsDir,
+			}
+		})
+
+		AfterEach(func() {
+			err = gcloud.DeleteBucket(testBucketName)
+			if err != nil {
+				_, _ = fmt.Fprintf(GinkgoWriter, "%s", err.Error())
+			}
+		})
+
+		It("fails to install vault with insufficient parameters", func() {
+			testParam := VaultCreationParam{}
+			err = vaultCreatorUnderTest.CreateOrUpdateVault(testParam)
+			Expect(err).ShouldNot(BeNil())
+
+			aggregate, ok := err.(errors.Aggregate)
+			Expect(ok).Should(BeTrue())
+			Expect(len(aggregate.Errors())).Should(Equal(5))
+
+			var errorMessages []string
+			for _, validationError := range aggregate.Errors() {
+				errorMessages = append(errorMessages, validationError.Error())
+			}
+			Expect(errorMessages).Should(ContainElement("the Vault name needs to be provided"))
+			Expect(errorMessages).Should(ContainElement("the namespace to create the Vault instance into needs to be provided"))
+			Expect(errorMessages).Should(ContainElement("a kube client needs to be provided"))
+			Expect(errorMessages).Should(ContainElement("a vault operator client needs to be provided"))
+			Expect(errorMessages).Should(ContainElement("a kube/cloud provider needs be provided"))
+		})
+
+		It("successfully installs vault", func() {
+			if cluster.IsInCluster() {
```

Comment:
Unfortunately, this test only works locally. The pipeline service account does not have enough permissions to run this test. Not sure what's the best way forward here. Would be nice to have these type of tests running in the cluster as well, but elevating the pipeline user's privileges might be a security issue. @ccojocar @daveconde